seed_data/management/commands/alter_data.py

```diff
@@ -175,32 +182,50 @@ def clear_future_runs(user=None, course=None):  # pylint: disable=unused-argumen
     return CourseRun.objects.filter(course=course, edx_course_key__startswith=NEW_COURSE_RUN_PREFIX).delete()
 
 
-@accepts_or_calculates_now
-def add_past_failed_run(user=None, course=None, now=None, grade=DEFAULT_FAILED_GRADE, add_if_exists=False):
-    """Adds a past failed course run for a given user and course"""
+def get_past_course_run(user=None, course=None, now=None, add_if_exists=False):
+    """Loop through past course runs and find one without CachedCurrentGrade data"""
     chosen_past_run = None
     past_runs = CourseRun.objects.filter(
         course=course,
         end_date__lt=now,
     ).exclude(end_date=None).order_by('-end_date').all()
-    # Loop through past course runs and find one without CachedCurrentGrade data
     for past_run in past_runs:
-        if not CachedCurrentGradeHandler(user).exists(past_run):
+        if not (CachedCurrentGradeHandler(user).exists(past_run)
+                or CachedCertificateHandler(user).exists(past_run)):
```

Comment:
I don't think removing continue will do it. I think you would need to either break out of the loop one you found the right course run or order the `CourseRun`s in the positive direction `.order_by('-end_date')`, so that the last course run it checks is the most recent one.